spec/controllers/homebuyings_controller_spec.rb

```diff
@@ -1,5 +1,160 @@
+        before :each do
+          sign_in @some_user
+        end
+        it "assigns a client id based on the id param" do
+          post :create, id: @some_other_client.id
+          expect(assigns(:id)).to eq @some_other_client.id
+        end
+      end
+    end
+
+    context "neither user or client logged in" do
+      it "doesnt assign id if no user or client is logged in" do
+        post :create
+        expect(assigns(:id)).to eq nil
+      end
+    end
+
+    describe "creates a new homebuying instance and assigns it to homebuying" do
+      it "assigns a homebuying instance" do
+        get :create
+        expect(assigns(:homebuying)).to be_a_new(Homebuying)
+      end
+    end
+
+    describe "it attempts to save the homebuying instance" do
+      context "saves successfully" do
+        before :each do
+          @this_client = create(:client)
+          sign_in @this_client
+        end
+        it "creates a new programemployee" do
+          expect{
+            post :create, attributes_for(:homebuying)
+          }.to change(ProgramEmployee, :count).by(1)
+        end
+        it "updates the flash hash with succuss message" do
+          post :create, attributes_for(:homebuying)
+          expect(flash[:success]).to be_present
+        end
+        it "redirects to the client status page" do
+          post :create, attributes_for(:homebuying)
+          expect(response).to redirect_to("/clients/#{@this_client.id}/status")
+        end
+      end
+
+      context "fails to save" do
+        before :each do
+          @this_client = create(:client)
+          sign_in @this_client
+          Homebuying.any_instance.stub(save: false)
+          Homebuying.any_instance.stub_chain(:errors, :full_messages).and_return(["danger"])
+        end
+        it "updates the flash message to danger" do
+          post :create, attributes_for(:homebuying)
+          expect(flash[:danger]).to be_present
+        end
+        it "renders the NEW template" do
+          post :create, attributes_for(:homebuying)
+          expect(response).to render_template :new
+        end
+      end
+    end
+  end
+
+  describe "edit #GET" do
+    context "client signed in" do
+      it "assigns a homebuying to @homebuying" do
+        some_client = create(:client)
+        some_homebuying = create(:homebuying, client_id: some_client.id)
+        sign_in some_client
+        get :edit, id: some_homebuying
+        expect(assigns(:homebuying)).to eq(some_client.homebuying)
+      end
+    end
+    context "user signed in" do
+      it "assigns a homebuying to @foreclosure" do
+        some_user = create(:user)
+        some_homebuying = create(:homebuying)
+        sign_in some_user
+        get :edit, id: some_homebuying
+        expect(assigns(:foreclosure)).to eq(some_homebuying)
+      end
+    end
+    it "renders the #edit template" do
+      get :edit, id: create(:homebuying)
+      expect(response).to render_template :edit
+    end
```

Comment:
Cool, I see you added the `edit.html.erb` file to get this test to pass. But it's never used in the app?